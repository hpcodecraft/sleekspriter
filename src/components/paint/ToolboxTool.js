import React from "react";

const ToolboxTool = (props) => {
  let cssClasses = "ToolBoxTool transparent";
  if(props.selected) cssClasses+= " active";

  return (
    <button
      id={props.id}
      className={cssClasses}
      disabled={props.selected}
      title={props.title}
      onClick={props.toolSelect.bind(this, props.id)}>
        <i className={props.icon}></i>
    </button>
  );
};

export default ToolboxTool;
