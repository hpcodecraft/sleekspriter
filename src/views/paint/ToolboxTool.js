import React from 'react';

const ToolboxTool = (props) => {
  let cssClasses = 'ToolBoxTool transparent';
  if(props.selected) cssClasses+= ' active';

  return (
    <button
      id={props.id}
      className={cssClasses}
      disabled={props.selected}
      onClick={props.selectTool.bind(this, props.id)}>
        <i className={props.icon}></i>
    </button>
  );
};

export default ToolboxTool;
