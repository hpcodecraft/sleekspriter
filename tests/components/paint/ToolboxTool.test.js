import ToolboxTool from "components/paint/ToolboxTool";

describe("ToolboxTool", () => {
  let props, wrapper;
  beforeEach(() => {
    props = {
      id: "BrushTool",
      icon: "flaticon-small23",
      selected: false,
      title: "Brush Tool (b)",
      toolSelect: jest.fn()
    };

    wrapper = shallow(<ToolboxTool {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render correctly when selected", () => {
    props.selected = true;
    wrapper = shallow(<ToolboxTool {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("is disabled when selected", () => {
    props.selected = true;
    wrapper = shallow(<ToolboxTool {...props} />);
    expect(wrapper.prop("disabled")).toEqual(true);
  });

  it("has the .active class when selected", () => {
    props.selected = true;
    wrapper = shallow(<ToolboxTool {...props} />);
    expect(wrapper.hasClass("active")).toEqual(true);
  });

  it("fires the toolSelect handler when clicked", () => {
    wrapper.simulate("click");
    expect(props.toolSelect).toBeCalledWith(props.id);
  });
});
