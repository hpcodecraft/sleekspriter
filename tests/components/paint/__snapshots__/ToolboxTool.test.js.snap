// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolboxTool should render correctly 1`] = `"<button id=\\"BrushTool\\" class=\\"ToolBoxTool transparent\\" title=\\"Brush Tool (b)\\"><i class=\\"flaticon-small23\\"></i></button>"`;

exports[`ToolboxTool should render correctly when selected 1`] = `"<button id=\\"BrushTool\\" class=\\"ToolBoxTool transparent active\\" disabled=\\"\\" title=\\"Brush Tool (b)\\"><i class=\\"flaticon-small23\\"></i></button>"`;
